### ISSUE TITLE
JSaddle XHR spin loop with "ob run"

### ISSUE THREAD

**Issue by mightybyte** [obsidiansystems/obelisk#874]
I'm running into a situation where JSaddle gets into an infinite loop and the app becomes unresponsive when I'm using `ob run`.  Steps to reproduce:

```
git clone https://github.com/kadena-io/block-explorer.git
cd block-explorer
git checkout jsaddle-spin-loop
ob run
```

Then, open a new browser window and before you go to the site, open the developer tools for that window and open the Network tab.  Then point the browser window at `localhost:8000`.  When I do this, some of the page loads and then it gets into a spin loop where hundreds of jsaddle XHR requests are sent and the app becomes unresponsive.  When I build and deploy the actual GHCJS app, the spin loop does not happen.  This happens to me on macOS Catalina using both Brave and Chrome.

I have done some investigation into this issue and have found the following interesting bits of information.  If I comment out line 488 in Frontend.hs (https://github.com/kadena-io/block-explorer/blob/f2d24653ca5d21d3319be98f730159ea464de86b/frontend/src/Frontend.hs#L488), the problem goes away.  However, if I comment out line 486 and replace it with line 487 (i.e. adding two `traceDyn` calls), the spin loop comes back again even though the `spacerRow` widget on line 488 is still commented out.

I've also encountered this spin loop behavior in a number of different obelisk apps that I've written (including https://github.com/kadena-io/chainweaver).  My anecdotal observation has been that it usually seems to be related to when my app makes XHR requests (not the jsaddle XHRs, but actual XHRs from my app).  The most recent time I encountered this it was in a new app and the spin loop started happening right when I introduced some XHRs to fetch data from a server.

**Comment by madeline-os** [comment]
Hi, is this still happening? If you could run with `ob run -v` and paste the log, that would be very helpful. Is this on darwin?

**Comment by mightybyte** [comment]
Yep, just encountered this issue again on a relatively new project.

**Comment by augyg** [comment]
@mightybyte Could you provide both sets of logs? (var/log/system.log and the app's console output) from my investigation so far there's a number of different issues that this could be related to

**Comment by augyg** [comment]
And to confirm, you are *still* on Catalina in the most recent attempt? Which would definitely not help to explain the launchd error as I might expect that type of error on Big Sur or Monterey.

**Comment by mightybyte** [comment]
This was run on macOS Catalina 10.15.7.

**Comment by augyg** [comment]
@mightybyte I've been able to reproduce similar behavior but not verbatim what you have here. I've been able to reproduce the behaviour with hundreds of jsaddle requests. 

Does this line 

`<interactive>: Network.Socket.sendBuf: resource vanished (Broken pipe)`

Occur consistently? 

Are you also using any other tooling such as nix-darwin or any system-wide nixpkgs overrides?

It seems weird to me that nixos.activate-system would be getting called. Exit code 126 means that its not an executable. This also makes me wonder if you're using nix-darwin and if that could be causing a difference here.

**Comment by mightybyte** [comment]
@augyg Yes, it does seem to occur consistently when I close the tab that has hung in the spin loop.

Yes, I am using nix-darwin.
